### DISCUSSION TITLE
Extend the req.auth object in Next.js middleware.

### DISCUSSION THREAD

**Discussion by space-symbol** [nextauthjs/next-auth#12097] [Help]
Is it possible to configure an `auth` object that is passed to the middleware function within the `request`? I haven't found any information about this in the documentation.

```ts
// middleware.ts 
const { auth: middleware } = NextAuth(authConfig);

export default middleware((req) => {
 console.log(req.auth) // { user: { name: "", email: "", image: "" }, expires: "" }
 
// other stuff...
}
```

**Comment by k-taro56** [comment]
You can find it in the documentation 😊

https://authjs.dev/reference/nextjs#authorized
